src/backend/distributed/deparser/ruleutils_13.c

```diff
@@ -775,13 +773,13 @@ has_dangerous_join_using(deparse_namespace *dpns, Node *jtnode)
 			 * dangerous situation and must pick unique aliases.
 			 */
 			RangeTblEntry *jrte = rt_fetch(j->rtindex, dpns->rtable);
-			ListCell   *lc;
 
-			foreach(lc, jrte->joinaliasvars)
+			/* We need only examine the merged columns */
+			for (int i = 0; i < jrte->joinmergedcols; i++)
 			{
-				Var		   *aliasvar = (Var *) lfirst(lc);
+				Node	   *aliasvar = list_nth(jrte->joinaliasvars, i);
 
-				if (aliasvar != NULL && !IsA(aliasvar, Var))
+				if (!IsA(aliasvar, Var))
```

Comment:
As far as I can tell this (and some more changes in this commit) is to reflect a change in upstream postgres https://github.com/postgres/postgres/commit/9ce77d75c5ab094637cc4a446296dc3be6e3c221

I think it will be good to reflect that in this commit message.

I think this is what you meant with rewriting this commit in our conversations.

Reply:
Yes that is what I meant since currently the commit message is completely wrong :D